tap_github/streams.py

```diff
@@ -375,3 +375,112 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class CommitsStream(GitHubStream):
+    """
+    Defines the 'Commits' stream.
+    The stream is fetched per repository to maximize optimize for API quota
+    usage.
+    """
+
+    name = "commits"
+    path = "/repos/{org}/{repo}/commits"
+    primary_keys = ["node_id"]
+    replication_key = "commit_timestamp"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    ignore_parent_replication_key = True
```

Comment:
Setting this to `true` means we don't expect the parent (repos, in this case) to have their update timestamp reflective of these additions. Is that correct?

To be fair, the `True` setting is the safest way, since we know we won't miss any updates, but I thought github included this in the repository's last updated timestamp. Maybe my misunderstanding is that only master/main-branch updates get reflected there?

Reply:
Thanks for clarifying the meaning of this. Just tested this by pushing a single commit to a repo on a throwaway branch:
- `pushed_at` gets updated to the timestamp of the commit
- `updated_at` is not modified.
If pushing to `master/main`, then `updated_at` does get updated as well, but there are a few seconds of gap between these values and the commit time (like this: commit time < pushed_at < updated_at), so on a very active project, I imagine this could lead to records slipping through depending on which timestamp we use.

Reply:
Okay, thanks for digging on this.

Based on your explanation, I think you have it correct that we should probably not count on this being authoritative at the parent level. (Keeping as-is to ignore=True, I think, is best.)